src/graph/FetchVerticesExecutor.cpp

```diff
@@ -93,35 +48,272 @@ Status FetchVerticesExecutor::prepareVids() {
+                        auto *column = new YieldColumn(expr);
+                        yieldColsHolder_.addColumn(column);
+                        yields_.emplace_back(column);
+                        colNames_.emplace_back(column->toString());
+                        colTypes_.emplace_back(nebula::cpp2::SupportedType::UNKNOWN);
+                        expCtx_->addInputProp(prop);
+                    }
+                    continue;
+                }
+            }
+
+            yields_.emplace_back(col);
+            col->expr()->setContext(expCtx_.get());
+            Status status = col->expr()->prepare();
+            if (!status.ok()) {
+                return status;
+            }
+            if (col->alias() == nullptr) {
+                colNames_.emplace_back(col->expr()->toString());
+            } else {
+                colNames_.emplace_back(*col->alias());
+            }
+            auto type = calculateExprType(col->expr());
+            colTypes_.emplace_back(type);
+            VLOG(1) << "type: " << static_cast<int64_t>(colTypes_.back());
+        }
+        if (expCtx_->hasSrcTagProp() || expCtx_->hasDstTagProp()) {
             return Status::SyntaxError(
-                "Near [%s.%s], tag should be declared in `ON' clause first.",
-                    pair.first.c_str(), pair.second.c_str());
+                "tag.prop and edgetype.prop are supported in fetch sentence.");
         }
 
-        if (labelSchema_->getFieldIndex(pair.second) == -1) {
-            return Status::Error("`%s' is not a prop of `%s'",
-                    pair.second.c_str(), pair.first.c_str());
+        auto aliasProps = expCtx_->aliasProps();
+        for (auto &pair : aliasProps) {
+            auto& tagName = pair.first;
+            auto& prop = pair.second;
+            if (tagNameSet_.find(tagName) == tagNameSet_.end()) {
+                return Status::SyntaxError(
+                    "Near [%s.%s], tag should be declared in `ON' clause first.",
+                    tagName.c_str(), prop.c_str());
+            }
+            auto tagStatus = ectx()->schemaManager()->toTagID(spaceId_, tagName);
+            if (!tagStatus.ok()) {
+                return tagStatus.status();
+            }
+            auto tagId = tagStatus.value();
+
+            std::shared_ptr<const meta::SchemaProviderIf> tagSchema =
+                ectx()->schemaManager()->getTagSchema(spaceId_, tagId);
+            if (tagSchema == nullptr) {
+                return Status::Error("No tag schema for %s", tagName.c_str());
+            }
+            if (tagSchema->getFieldIndex(prop) == -1) {
+                return Status::Error(
+                    "`%s' is not a prop of `%s'", tagName.c_str(), prop.c_str());
+            }
+            storage::cpp2::PropDef pd;
+            pd.owner = storage::cpp2::PropOwner::SOURCE;
+            pd.name = prop;
+            pd.id.set_tag_id(tagId);
+            props_.emplace_back(std::move(pd));
```

Comment:
Where did your support allProps on one Tag.   Such as tag1.*

Reply:
tag1.* can not pass syntax check.

Reply:
Is is the compatible with the original behavior?

Reply:
the behavior is same.